Update JS Release Notes (#1291)
SAP/cloud-sdk: docs-js/release-notes.mdx

- [connectivity] The generic types of `JwtKeyMapping` is simplified so the second type argument `JwtKeysT` are always strings. (1b974095f)
- [connectivity] The `Protocol` enum was replaced with a string literal union type. Now, instead of passing `Protocol.HTTPS` pass 'https'. (1b974095f)
- [connectivity, http-client, openapi, odata-common, odata-v2, resilience] Remove the options `timeout` and `enableCircuitBreaker` from all BTP service interactions i.e. destination and token fetching. (1b974095f)
- [connectivity, http-client, mail-client, openapi, odata-common, odata-v2, odata-v4, eslint-config, generator, test-util, util, openapi-generator, generator-common, temporal-de-serializers, resilience] Switch the compilerOptions.module to `Node16` instead of `CommonJS` for better interoperability with ES modules. See the [TypeScript handbook](https://www.typescriptlang.org/docs/handbook/esm-node.html) for technical details if needed. (1b974095f)
- [connectivity] The enum `IsolationStrategy` was replaced with a string literal union type of the same name. Use 'tenant' and 'tenant-user' instead of `IsolationStrategy.Tenant` and `IsolationStrategy.Tenant_User`. (1b974095f)

> - [connectivity, http-client, odata-common] ] The `url` property in `Destination` is now optional as destinations of type `MAIL` do not have a URL. (1b974095f)

Reviewer: ```suggestion
- [connectivity, http-client, odata-common] The `url` property in `Destination` is now optional as destinations of type `MAIL` do not have a URL. (1b974095f)
```

  The type of the parameter is now `Record<string, any>`, and if only want to set the `Content-Type` HTTP header you can do so by passing `{'content-type': 'some-value'}` to the constructor. (1b974095f)
- [odata-common] The constructor of the entity API is private to avoid accidental usage. Use the service object to get a API instance. (1b974095f)
- [odata-common, generator, openapi-generator] The `serviceMapping` option of the OData generator has been renamed to `optionsPerService`. The mapping file, `service-mapping.json` has also been renamed to `options-per-service.json`. By default, an options file will not be generated. (1b974095f)
- [odata-common] Setting custom fields in `fromJson` through the `_customFields` property has been removed.
  Add custom properties to your JSON object instead. (1b974095f)
- [generator] The options per service behavior is now the same as for the OpenApi generator.
  For details on how to migrate and what has change look at the migration guide. (1b974095f)
- [generator] The deprecated `forceOverwrite` option of the generator is removed. Use the `overwrite` option instead.

>   [Compatibility Note] The deprecated `generatePackageJson` option of the generator is removed. Use the `packageJson` option instead.

Reviewer: ```suggestion
- [generator] The deprecated `generatePackageJson` option of the generator is removed. Use the `packageJson` option instead.
```

>   [Compatibility Note] The deprecated `writeReadme` option of the generator is removed. Use the `readme` option instead.

Reviewer: ```suggestion
- [generator] The deprecated `writeReadme` option of the generator is removed. Use the `readme` option instead.
```

>   [Compatibility Note] The deprecated `processesJsGeneration` option of the generator is removed. Use the `transpilationProcesses` option instead.

Reviewer: ```suggestion
- [generator] The deprecated `processesJsGeneration` option of the generator is removed. Use the `transpilationProcesses` option instead.
```

>   [Compatibility Note] The internal options `sdkAfterVersionScript`, `s4HanaCloud` and `packageVersion` of the generator are removed.

Reviewer: ```suggestion
- [generator] The internal options `sdkAfterVersionScript`, `s4HanaCloud` and `packageVersion` of the generator are removed.
```

  These were hidden options never meant for external usage and there is no replacement.

>   [Compatibility Note] The deprecated `generateNpmrc` option of the generator is removed. Use the `include` option to add a `.npmrc` to the generated code if needed. (1b974095f)

Reviewer: ```suggestion
- [generator] The deprecated `generateNpmrc` option of the generator is removed. Use the `include` option to add a `.npmrc` to the generated code if needed. (1b974095f)
```

- [generator] Add `verbose` generator option. By default, only error and warning logs will be displayed. If `verbose` is set to `true`, all logs will be displayed. (1b974095f)
- [generator] The `generateJs` option has been replaced with the `transpile` option. Transpilation is not enabled by default. (1b974095f)
- [generator] The command line argument `inputDir` of the OData generator is renamed to `input`.
  The new `input` options accepts now also file paths and glob patterns. (1b974095f)
- [generator] The OData generator won't automatically rename identifiers to avoid name conflicts.
  The generation process will fail if identifiers have conflicting names.
  Switch on the `skipValidation` flag if you want to generate despite name conflicts and are okay with changed identifier names to avoid conflicts. (1b974095f)
- [generator, openapi-generator, generator-common] Removed the option `versionInPackageJson` from the OData and OpenAPI generator.
  If the option `packageJson` is enabled now a `package.json` with a default version `1.0.0` is created.
  If necessary use the `include` option to add a `package.json` with a custom value.

>   [Compatibility Note] Removed the option `licenseInPackageJson` from the OData and OpenAPI generator.

Reviewer: ```suggestion
- [generator, openapi-generator] Removed the option `licenseInPackageJson` from the OData and OpenAPI generator.
```

- [generator] The option `generateCSN` is removed. There is no replacement. (1b974095f)
- [generator] The type for paths in the `GeneratorOptions` is changed from `fs.PathLike` to `string`.
  In case you passed a buffer object please resolve it to a string before passing it to the SAP Cloud SDK. (1b974095f)
- [generator, openapi-generator, generator-common] The two generators use the same CLI parsing code now, aligning the way paths are resolved.
  In case you experience problems with the new logic, enable the `verbose` flag to investigate the new paths. (1b974095f)
- [util] The function `variadicArgumentToArray` is replaced by the function `transformVariadicArgumentToArray`. (1b974095f)
- [util] The field `logger` on the interface `LoggerOptions` was not used and is removed from the interface. (1b974095f)
- [openapi-generator] The internal option `packageVersion` of the OpenAPI generator is removed. (1b974095f)
- [openapi-generator] `tsConfig` option has been renamed to `tsconfig`. (1b974095f)
- [generator-common] The generator does not create `d.ts.map` files per default anymore. If you need them include a custom `tsconfig.json`.

>   [Compatibility Note] All SAP Cloud SDK packages ship without `d.ts.map` files from now on.

Reviewer: ```suggestion
- [connectivity, http-client, mail-client, openapi, odata-common, odata-v2, odata-v4, eslint-config, generator, test-util, util, openapi-generator, generator-common, temporal-de-serializers, resilience] All SAP Cloud SDK packages ship without `d.ts.map` files from now on.
```
